app/src/main/java/org/oppia/android/app/home/HomeViewModel.kt

```diff
@@ -107,20 +116,39 @@ class HomeViewModel(
   }
 
   /**
-   * Returns a [HomeItemViewModel] corresponding to the promoted stories to be displayed for this learner
-   * (see [PromotedStoryListViewModel]), or null if this profile does not have any promoted stories.
-   * Promoted stories are determined by any recent stories started by this profile.
+   * Returns a [HomeItemViewModel] corresponding to the promoted stories[PromotedStoryListViewModel] and Upcoming topics
+   * [ComingSoonTopicListViewModel]  to be displayed for this learner or null if this profile does not have any promoted stories.
+   * Promoted stories are determined by any recent stories or suggested stories started by this profile.
    */
-  private fun computePromotedStoryListViewModel(
-    ongoingStoryList: OngoingStoryList
+  private fun computeRecommendedActivityListViewModel(
+    recommendedActivityList: RecommendedActivityList
   ): HomeItemViewModel? {
-    val storyViewModelList = computePromotedStoryViewModelList(ongoingStoryList)
-    return if (storyViewModelList.isNotEmpty()) {
-      return PromotedStoryListViewModel(
-        activity,
-        storyViewModelList
-      )
-    } else null
+    when (recommendedActivityList.recommendationTypeCase) {
+      RecommendedActivityList.RecommendationTypeCase.RECOMMENDED_STORY_LIST -> {
```

Comment:
Is RecommendationTypeCase is an enum ? if yes then we don't need to handle the else case here

Reply:
Its not an enum but oneOf at most one field will be set at the same time.

Reply:
Protobuf enum can, unfortunately, be unspecified so we can't rely on exhaustive when statements. :( Hopefully this gets addressed when there's a Kotlin-compatible protobuf compiler.